lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -617,6 +617,25 @@ defmodule ExDoc.Formatter.HTML.Autolink do
     end
   end
 
+  defp hex_lib_dirs() do
+    [{hex_ebin_path(), @elixir_docs <> "hex/"}]
+  end
+
+  defp hex_ebin_path() do
+    # TODO: Private Elixir API
+    # TODO: Not available when running ex_doc as escript
+    archive_path = Mix.Local.path_for(:archive)
+
+    hex_archive =
+      archive_path
+      |> Path.join("hex-*")
+      |> Path.wildcard()
+      |> List.first()
```

Comment:
Path.wildcard sorts lexiographically so if we pick last here we should get a later version (although it's unlikely to have multiple versions installed).